src/proxy/http/add_header.rs

```diff
@@ -0,0 +1,229 @@
+use std::{fmt, marker::PhantomData};
+
+use http::header::{AsHeaderName, HeaderValue};
+
+use svc;
+
+type Ret<T> = fn(&T) -> Option<HeaderValue>;
```

Comment:
This probably warrants a description.  Also, I have no idea what `Ret` is supposed to be short for. Can this be an actual name?